Contact form plugin (#2191)
MarkBind/markbind: docs/userGuide/plugins/web3Form.md

### Plugin: Web3Forms 

This plugin allows you to create forms whose response will be sent directly to your email, using the [Web3Form](https://web3forms.com/) API. 
<box type="warning" seamless>

> Emails from Web3Forms may end up in spam/junk mail.

Reviewer: Thanks for adding this in! Could we rephrase this as an actionable tip for users? e.g. "Don't forget to check your spam/junk folder if your form responses aren't appearing in your inbox!" 
We could also use a `tip` type for the `<box>` to reduce the number of consecutive warning messages :)

</box>

<box type="warning" seamless>

> The free plan from Web3Form only allows 250 submissions per month.

Reviewer: ```suggestion
The free plan from Web3Forms only allows 250 submissions per month.
```
Nit, can we standardise to use "Web3Forms" throughout? Same for the rest of the docs

Reviewer: ```suggestion
The free plan from Web3Form allows up to 250 submissions per month.
```
Another nit, to sound more optimistic 😄

### Plugin: Web3Forms 

This plugin allows you to create forms whose response will be sent directly to your email, using the [Web3Form](https://web3forms.com/) API. 

> To set it up, get an access key from [Web3Forms](https://web3forms.com/). Then add`web3Form` to your site's plugin, and add the `accessKey` parameter via the `pluginsContext`.

Reviewer 2: @yucheng11122017 I assume this access key is not a 'secret' key?

Author: Urm not exactly? It's linked to the user's email. web3forms uses it to redirect the form submissions to the user's email.

Reviewer: Is there potential for misuse of the key? My current understanding is that if someone wants to abuse this key, they could use it to direct the submissions from a different form to the user's email address, but this would have the same effect as simply spamming junk submissions to the original form so it doesn't give a malicious user any additional tool for abuse. 
Unless the key can be used to reveal the user's actual email (which the user might have intended to keep private)?

Reviewer 2: Should be fine, if the key is not meant to be a 'secret' key that should not be made public.

| Name      | Type     | Default                             | Description |
|-----------|----------|-------------------------------------|-------------|
| accessKey | `String` ||  accessKey is provided by Web3Form. |

```js {heading="site.json"}
{
  ...
  "plugins": [
    "web3Form"
  ],
  "pluginsContext": {
    "web3Form": {

>       "accesskey": "YOUR_WEB_3_FORM_ACCESS_KEY", // replace with your Web3Form access key 

Reviewer: Also is this code block language is labelled `js` instead of `json` to support comments? 🤔

>       "accesskey": "YOUR_WEB_3_FORM_ACCESS_KEY", // replace with your Web3Form access key 

Author: Opps sorry I copied it from documentation on other plugins which was using js.

>       "accesskey": "YOUR_WEB_3_FORM_ACCESS_KEY", // replace with your Web3Form access key 

Author: I believe so. Changing it to JSON will get warnings regarding comments and regarding the trailing commas in the start

>       "accesskey": "YOUR_WEB_3_FORM_ACCESS_KEY", // replace with your Web3Form access key 

Reviewer: Ah I see, in that case I think it's fine to keep it as `js` to maintain consistency with the other plugins especially since the difference in syntax highlighting would be visible to users

### Plugin: Web3Forms 

This plugin allows you to create forms whose response will be sent directly to your email, using the [Web3Form](https://web3forms.com/) API. 
<box type="warning" seamless>

> Emails from Web3Forms often end up in spam / junk mail.

Reviewer 3: ```suggestion
Emails from Web3Forms may end up in spam/junk mail.
```

</box>

<box type="warning" seamless>
The free plan from Web3Form only allows 250 submissions per month.
</box>

> To set it up, get an access key from [Web3Forms](https://web3forms.com/). Then add `web3Form` to your site's plugin, and add the `accessKey` parameter via the `pluginsContext`.

Reviewer 3: We should mention whether the access key is ok to be exposed or at least make known that it is exposed, given that it's a working form without a backend. I tried looking at the web3forms docs and it doesn't seem to mention it at all. Perhaps it's due to the fact that the project seems really new...... Nonetheless, we should do a bit of research and put a note about this in our docs, explaining:
- whether this key is exposed to public
- whether such exposure is risky and has any consequences

Author: The key will be exposed to the public since users can view the API call to web3forms and see the access key 
ie Developer tools -> network -> payload 
![image](https://user-images.githubusercontent.com/77204346/223029160-941a947e-8c2c-422b-ab6c-07e24f1653f2.png)
It is [mentioned](https://docs.web3forms.com/getting-started/installation#:~:text=%3C!%2D%2D%20REQUIRED%3A%20Your%20Access%20key%20here.%20Don%27t%20worry%20this%20can%20be%20public%20%2D%2D%3E) that its ok for the access key to be public. I emailed them to ask why and they said 

> Yes, it can be public.
> 
> The access key is an alias to your email. This is same as revealing email in public but one step harder. 
> 
> The people who get the access key can only used to email you. Nothing else. We also have spam protection + reCaptcha & domain protection in the Pro plan.

The problem with exposing the key is that other forms can use it and attackers might use it to DOS especially since only 250 submission are allowed per month as mentioned by @jovyntls. 

I will add the above to the documentation!

Reviewer 3: Is this in response to the question above:
![image](https://user-images.githubusercontent.com/41845017/226260242-0d82b181-a3c5-48b3-8538-0170628fc3ce.png)

?

Can we re-word it such that it says something along the line of
- generally ok to be public as shared by the team behind it (the reason you gave here)
- however, some potential problem is xxxx

This is because, at the moment, I'm not sure what to make of the warning after reading it...-> it seems to suggest that I shouldn't expose the access key given all the negatives

Author: Thanks for the comments @tlylt, updated in [eef1b38](https://github.com/MarkBind/markbind/pull/2191/commits/eef1b38665d42a54440dfd7bb1d401def31fef98)

| Name      | Type     | Default                                            | Description |
|-----------|----------|----------------------------------------------------|-------------|
| accessKey | `String` || accessKey is required. It is provided by Web3Form. |

```json {heading="site.json"}
{
  ...
  "plugins": [
    "web3Form"
  ],
  "pluginsContext": {
    "web3Form": {
      "accessKey": "YOUR_WEB_3_FORM_ACCESS_KEY", // replace with your Web3Form access key 
    }
  }
}
```

**To add a form to your site, use a `<web-3-form>` tag.**

<include src="codeAndOutput.md" boilerplate>
<variable name="highlightStyle">html</variable>
<variable name="code">
<web-3-form header="Contact Us">
    <label for="name">Name</label>
    <input type="text" name="name" required placeholder="John Doe">
    <button type="submit">Submit</button>
</web-3-form>
</variable>
</include>

**Web3Form plugin also supports a default 'Contact Us' form.**

<include src="codeAndOutput.md" boilerplate>
<variable name="highlightStyle">html</variable>
<variable name="code">

> <web-3-form default>
> </web-3-form>

Reviewer 3: ```suggestion
<web-3-form default></web-3-form>
```

Does this work if we write `<web-3-form default />` ? If so perhaps we can simplify this and other similar cases in the docs.

</variable>
</include>

**Web3Form plugin supports common inputs.**

The inputs supported are 
* `name-input`
* `email-input`
* `message-input`
* `submit-button`

<include src="codeAndOutput.md" boilerplate>
<variable name="highlightStyle">html</variable>
<variable name="code">
<web-3-form header="### Form with name input">
<name-input></name-input>
</web-3-form>

<web-3-form header="### Form with email input">
<email-input></email-input>
</web-3-form>

<web-3-form header="### Form with message input">
<message-input></message-input>
</web-3-form>

<web-3-form header="### Form with submit button">
<submit-button></submit-button>
</web-3-form>

</variable>
</include>

**Web3Form plugin supports customised header and styles**

> Options from the [box component](https://markbind.org/userGuide/components/presentation.html#boxes) are supported here.

Reviewer 3: use a relative reference to avoid linking out of the currently deployed site, i.e from deploy preview to markbind.org, or from markbind-master to markbind.org
